src/progress/progress_set.rs

```diff
@@ -185,13 +190,34 @@ impl ProgressSet {
                 DefaultHashBuilder::default(),
             ),
             sort_buffer: RefCell::from(Vec::with_capacity(voters)),
-            configuration_capacity: (voters, learners),
             configuration: Configuration::with_capacity(voters, learners),
             next_configuration: Option::default(),
             logger: default_logger().new(o!()),
         }
     }
 
+    pub(crate) fn restore_conf_states(
```

Comment:
Strange method name. I would expect it mutate self instead of returning a new instance.